logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -111,10 +111,15 @@ private enum class CommitStrategy {
 }
 
 private fun CoreFailure.getStrategy(
+    retryCount: Int,
```

Comment:
question: Is `retryCount` the number of tries until now, or the maximum number of tries?

It looks like number of tries, where it starts with 2 and goes down as it fails. The name + `retryCount > 0` looks a bit weird.

The logic is a bit split between `getStrategy` and `handleCommitFailure`, which makes a bit harder to follow and figure out what's going on.

Maybe renaming it to `remainingAttemptsCount`, so it's clearer that if it's `0` we should stop.

Reply:
Yep confusing name, I'll rename it.

>The logic is a bit split between getStrategy and handleCommitFailure, which makes a bit harder to follow and figure out what's going on.

Any idea on how to improve it? Given the current recursive setup I don't see how avoid also having a parameter in the `handleCommitFailure` function.

Reply:
> Yep confusing name, I'll rename it.
> 
> > The logic is a bit split between getStrategy and handleCommitFailure, which makes a bit harder to follow and figure out what's going on.
> 
> Any idea on how to improve it? Given the current recursive setup I don't see how avoid also having a parameter in the `handleCommitFailure` function.

Not really. I think the name change was already an improvement. Made it clearer.